napari/layers/labels/labels.py

```diff
@@ -340,6 +352,45 @@ def selected_label(self, selected_label):
         self._selected_color = self.get_color(selected_label)
         self.events.selected_label()
 
+    @property
+    def enable_custom_color_dict(self):
+        """True if enabling custom color dictionary"""
+        return (
+            hasattr(self, '_enable_custom_color_dict')
+            and self._enable_custom_color_dict
+        )
+
+    @enable_custom_color_dict.setter
+    def enable_custom_color_dict(self, enable_custom_color_dict):
+
+        if not self.color_dict:
+            return
+
+        if enable_custom_color_dict:
+            self.custom_color = {0: 0.0}
```

Comment:
Can we move this to a function in https://github.com/napari/napari/blob/master/napari/utils/colormaps/colormaps.py, maybe `color_dict_to_colormap` that takes in the color dict and any other necessary args, like `custom_color` and then returns a vispy colormap. This file should not need to import either vispy `Colormap` or `Color` but should just import that function. We should also be able to test that function independently of the rest of the codebase.

@kevinyamauchi might also have tips on how the `Color` transformations are done, we might have some utilities that he knows of that will make that easier, maybe he can have a look here too.

Reply:
We have been trying to move to representing individual colors as RGBA arrays. @HagaiHargil made some nice utilities for converting common color representations to RGBA arrays. Some that you may be interested in:

- [transform_color()](https://github.com/napari/napari/blob/e9df14f6c6f09641faa0e2b41f57a0d55632f7f0/napari/utils/colormaps/standardize_color.py#L31): transform a single color (e.g., common string name, RGB) into an RGBA array
- [transform_color_with_defaults()](https://github.com/napari/napari/blob/e9df14f6c6f09641faa0e2b41f57a0d55632f7f0/napari/layers/utils/color_transformations.py#L20): calls `transform_color()` but you can provide fallback values
- [normalize_and_broadcast_colors()](https://github.com/napari/napari/blob/e9df14f6c6f09641faa0e2b41f57a0d55632f7f0/napari/layers/utils/color_transformations.py#L91): broadcast a color into an array of colors with the specified number of elements.

Reply:
I was originally trying to go with RGBA array representation, but it seems beyond the scope, because I need to change how the label data layer is represented, which is in the colormap and float between [0,1], we would need to go very deep to change the entire flow to work with RGBA arrays, which felt like a major refactor

Reply:
Okay, that makes sense to me. I think there will be colormap overhaul in the not-so-distant future, so we can address this then.